File: dailyChallenges/0008-BestTimeToBuyAndSellStock/8-BestTimeToBuyAndSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        # Buy = l
        # Sell = r

        l , r = 0, 0
        maxProfit, profit = 0, 0

        while r < len(prices):
            if prices[l] < prices[r]:
                profit = prices[r] - prices[l]
                maxProfit = max(maxProfit, profit)
            else:
                l = r
            r += 1
        return maxProfit

File: dailyChallenges/0008-BestTimeToBuyAndSellStock/8-BestTimeToBuyAndSellStock.go
func maxProfit(prices []int) int {
	l, r := 0, 1
	maxProfit, profit := 0, 0

	for r < len(prices) {
		if prices[l] < prices[r] {
			profit = prices[r] - prices[l]
			maxProfit = max(maxProfit, profit)
		} else {
			l = r
		}
		r += 1
	}
	return maxProfit

}